### PR TITLE
s390x: add support for virtio dasds

### PR DESCRIPTION
**Pull request by nikita-dubrovskii**
This is a fix for https://bugzilla.redhat.com/show_bug.cgi?id=1960485

Signed-off-by: Nikita Dubrovskii <nikita@linux.ibm.com>

### CODE REVIEW

File: src/blockdev.rs [hunk 1]
```diff
@@ -970,10 +970,42 @@ pub fn detect_formatted_sector_size(buf: &[u8]) -> Option<NonZeroU32> {
...
+
+    if target.starts_with("/dev/vd") {
+        let cdl_magic = [0xd3, 0xf1, 0xe5, 0xd6];
+        let lbl;
+        if let Some(t) = fd {
+            lbl = read_magic(device, t)?;
+        } else {
+            let mut disk = File::open(device).with_context(|| format!("opening {}", device))?;
+            lbl = read_magic(device, &mut disk)?;
+        }
```
Comment (bgilbert) [inline src/blockdev.rs:1002]:
```suggestion
        let lbl = if let Some(t) = fd {
            read_magic(device, t)?
        } else {
            let mut disk = File::open(device).with_context(|| format!("opening {}", device))?;
            read_magic(device, &mut disk)?
        }
```

File: src/blockdev.rs [hunk 2]
```diff
@@ -970,10 +970,42 @@ pub fn detect_formatted_sector_size(buf: &[u8]) -> Option<NonZeroU32> {
 }
 
 /// Checks if underlying device is IBM DASD disk
-pub fn is_dasd(device: &str) -> Result<bool> {
+pub fn is_dasd(device: &str, fd: Option<&mut File>) -> Result<bool> {
     let target =
         canonicalize(device).with_context(|| format!("getting absolute path to {}", device))?;
-    Ok(target.to_string_lossy().starts_with("/dev/dasd"))
+    if target.starts_with("/dev/dasd") {
```
Comment (bgilbert) [inline src/blockdev.rs:976]: Do these checks work?  `PathBuf.starts_with()` is [documented](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.starts_with) to compare only whole path components.

File: src/blockdev.rs [hunk 3]
```diff
@@ -970,10 +970,42 @@ pub fn detect_formatted_sector_size(buf: &[u8]) -> Option<NonZeroU32> {
...
+        let cdl_magic = [0xd3, 0xf1, 0xe5, 0xd6];
+        let lbl;
+        if let Some(t) = fd {
+            lbl = read_magic(device, t)?;
+        } else {
+            let mut disk = File::open(device).with_context(|| format!("opening {}", device))?;
+            lbl = read_magic(device, &mut disk)?;
+        }
+        if cdl_magic == lbl {
+            eprintln!("detected virtblk ECKD CDL disk: {}", device);
```
Comment (bgilbert) [inline src/blockdev.rs:1004]: Let's not print from a utility function that's called multiple times.

**Review by bgilbert — COMMENTED** [review]
LGTM generally.

File: src/blockdev.rs (hunk 2 above)
Comment (nikita-dubrovskii) [reply src/blockdev.rs]: missed that part, fixed

**Review by bgilbert — APPROVED** [review]
:+1:

**Pull request merged**